repair_restrict_lot/views/repair_order_view.xml

```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!--
+  Copyright 2025 ForgeFlow S.L. (https://www.forgeflow.com)
+  License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+-->
+<odoo>
+    <record id="view_repair_order_form" model="ir.ui.view">
+        <field name="name">repair.order.form - repair_restrict_lot</field>
+        <field name="model">repair.order</field>
+        <field name="inherit_id" ref="repair.view_repair_order_form" />
+        <field name="arch" type="xml">
+            <xpath
+                expr="//field[@name='move_ids']/tree//field[@name='lot_ids']"
+                position="before"
+            >
+                <field
+                    name="restrict_lot_id"
+                    string="Lot/Serial Number"
+                    groups="stock.group_production_lot"
+                    domain="[('product_id','=',product_id)]"
```

Comment:
Maybe adding an invisible if the product is not tracked by lots makes sense: 
```
<field name="has_tracking" column_invisible="1"/>
<field name="restrict_lot_id" string="Lot/Serial Number" invisible="has_tracking == 'none'" groups="stock.group_production_lot" domain="[('product_id','=',product_id)]"/>

Reply:
good suggestion! Done